kwave/utils/mapgen.py

```diff
@@ -2863,6 +2863,117 @@ def calculate_lateral_pressure() -> NDArray[Shape["N"], Float]:
+    by axial_position. Where 0 corresponds to the transducer surface.
+
+    Note, O'Neil's formulae are derived under the assumptions of the
+    Rayleigh integral, which are valid when the transducer diameter is
+    large compared to both the transducer height and the acoustic
+    wavelength.
+
+    Example:
+        # define transducer parameters
+        radius = 140e-3  # [m]
+        diameter = 120e-3  # [m]
+        velocity = 100e-3  # [m / s]
+        frequency = 1e6  # [Hz]
+        sound_speed = 1500  # [m / s]
+        density = 1000  # [kg / m^3]
+
+        # define position vectors
+        axial_position = np.arange(0, 250e-3 + 1e-4, 1e-4)  # [m]
+        p_axial = focused_annulus_oneil(radius, diameter, amplitude, phase, frequency, sound_speed, density, axial_position)
+
+    Args:
+        radius: transducer radius of curvature [m]
+        diameter: 2 x num_elements array containing pairs of inner and outer aperture diameter
+                  (diameter of opening) [m].
+        amplitude: array containing the normal surface velocities for each element [m/s]
+        phase: array containing the phase for each element [rad]
+        frequency: driving frequency [Hz]
+        sound_speed: speed of sound in the propagating medium [m/s]
+        density: density in the propagating medium [kg/m^3]
+        axial_positions: vector of positions along the beam axis where the
+                         pressure amplitude is calculated [m]
+
+    Returns:
+        p_axial: pressure amplitude at the positions specified by axial_position [Pa]
+
+    References:
+        O'Neil, H. (1949). Theory of focusing radiators. J. Acoust. Soc. Am., 21(5), 516-526.
+
+    See also focused_bowl_oneil.
+    """
+
+    assert (diameter.ndim == 2), "wrong shape for diameter"
+    assert (2 in np.shape(diameter)), "wrong shape for diameter"
+    assert (np.greater_equal(diameter, np.zeros_like(diameter)).all() and np.isreal(diameter).all() and np.isfinite(diameter).all()), "wrong values in diameter object"
+
+    # set the number of elements in annular array
+    num_elements: int = [int(x) for x in np.shape(diameter) if x != 2][0]
+
+    assert (np.size(amplitude) == num_elements), "amplitude and num_elements"
+
+    assert (np.shape(amplitude) == np.shape(phase)), "amplitude and phase have different shapes"
+    assert (np.size(np.squeeze(amplitude)) == num_elements), "amplitude object does not the same size as elements"
+    assert (np.all(np.isreal(amplitude)) and np.all(np.isfinite(amplitude))), "amplitude has a bad value"
+    assert (np.all(np.isreal(frequency)) and np.all(np.isfinite(frequency))), "frequency has a bad value"
+
+    assert ((radius > 0.0) and np.isreal(radius) and np.isfinite(radius) ), "radius is incorrect"
+
+    assert ((phase >= 0.0).all() and (phase <= 2.0 * np.pi).all() and np.isreal(phase).all()
+            and np.isfinite(phase).all()), "phase is incorrect"
```

Comment:
Please check user input with if-else statements. See Farid's [previous comment ](https://github.com/waltsims/k-wave-python/pull/242#pullrequestreview-1815885992)for the reason:

 > If we want to take one extra step, we can convert assertions to raise (...) statements. It is recommended to use assertions for internal code logic only. User inputs should be checked with if-else statements. [This article](https://stefan.sofa-rockers.org/2018/04/16/assertions-and-exceptions/) explains a possible problem when using assertions to validate user inputs. There was another article that discussed this topic from code structure PoV but I can't find it.